nodebuilder/tests/api_test.go

```diff
@@ -37,7 +38,7 @@ func TestNodeModule(t *testing.T) {
 
 	sw := swamp.NewSwamp(t, swamp.WithBlockTime(time.Second))
 	// start a bridge node
-	bridge := sw.NewBridgeNode()
+	bridge := sw.NewBridgeNode(fx.Replace(sw.Fetcher()))
```

Comment:
any idea what a swamp is and why this needs to replace a "swamp fetcher"? 😂

Reply:
yeah that was the main gotcha in this PR, the `Fetcher` is a `BlockFetcher`, which maybe it should be renamed to, which uses an underlying comet/core grpc connection.

The problem is that unless it is replaced, a new instance is returned which does not use the correct gRPC connection and things blow up. 

There may be a nicer way to do this, but we can refactor the `fx.Replace` if we have a cleaner way!